### ISSUE TITLE
/done /b_alloc in supernova bad TCP osc 

### ISSUE THREAD

**Issue by sonoro1234** [supercollider/supercollider#4400]
When working with my SC client I could see an issue only happening in supernova:

With tcp connection, after sending a /b_alloc the response is malformed osc sometimes (one in 8 attemps more or less)

I dont know if sclang reports this kind of errors in received tcp.

I could read for example
```
"\0\0\0\28"/done"\0\0\0",si"\0"/b_alloc"\0\0\0\0\0\0\0"	/vum
```
As the response is garbled with other osc I think it is important to reproduce it having other osc responses arround as could be the generated by SendReply or /n_go messages after synth creation.

I had specifically SendPeakRMS responses and also `/b_setn` in response to `b_getn`
Also problematic /b_alloc with complection message creating a synth: n_go and /done /b_alloc get garbled.

If someone can confirm that issue, I suspect the cause could be some thread which takes care of /b_alloc not protected by mutex or whatever.

Could be related to #4062 althought there is not b_alloc there

Perhaps only @timblechmann could solve this.

**Comment by sonoro1234** [comment]
Also this happened
```
node not found or not a group
node not found or not a grException when sending message over TCP: oup
node not found or not a group
```
output is grabled which could reflect threads not being handled well. The exception does not report e.What.

**Comment by sonoro1234** [comment]
b_zero behaves well.
code is different:
b_alloc
```
                cmd_dispatcher<realtime>::fire_system_callback( [=]{
                    free_aligned(free_buf);
                    send_done_message(endpoint, "/b_alloc", bufferIndex);
                });
```
and b_zero
```
cmd_dispatcher<realtime>::fire_rt_callback( [ =, message = std::move(message) ] () mutable {
            sc_factory->increment_write_updates(index);
            handle_completion_message( std::move( message), endpoint );
            cmd_dispatcher<realtime>::fire_done_message( endpoint, b_zero, index );
        });
```
done message sent from system_callback or rt_callback

**Comment by sonoro1234** [comment]
Issue was solved in PR #4410 adding a mutex to the tcp send function so that can be called from different threads.
